maro/rl/policy/policy_manager.py

```diff
@@ -339,3 +339,193 @@ def exit(self):
+            minimum number of experiences in the experience memory required to trigger a call to ``learn`` for
+            each policy. Defaults to None, in which case all warm-up sizes will be set to 1.
+        post_update (Callable): Custom function to process whatever information is collected by each
+            trainer (local or remote) at the end of ``update`` calls. The function signature should be (trackers,)
+            -> None, where tracker is a list of environment wrappers' ``tracker`` members. Defaults to None.
+        log_dir (str): Directory to store logs in. A ``Logger`` with tag "POLICY_MANAGER" will be created at init
+            time and this directory will be used to save the log files generated by it. Defaults to the current
+            working directory.
+        proxy_kwargs: Keyword parameters for the internal ``Proxy`` instance. See ``Proxy`` class
+            for details. Defaults to the empty dictionary.
+    """
+    def __init__(
+        self,
+        policy_dict: Dict[str, AbsCorePolicy],
+        group: str,
+        num_trainers: int,
+        update_trigger: Dict[str, int] = None,
+        warmup: Dict[str, int] = None,
+        post_update: Callable = None,
+        log_dir: str = getcwd(),
+        proxy_kwargs: dict = {}
+    ):
+        super().__init__(policy_dict, update_trigger=update_trigger, warmup=warmup, post_update=post_update)
+        peers = {"trainer": num_trainers}
+        self._proxy = Proxy(group, "policy_manager", peers, component_name="POLICY_MANAGER", **proxy_kwargs)
+
+        self._policy2trainer = defaultdict(list)
+        self._trainer2policies = defaultdict(list)
+        self._exp_cache = defaultdict(ExperienceSet)
+        self._num_experiences_by_policy = defaultdict(int)
+        self.num_trainers = num_trainers
+
+        self._logger = Logger("MULTINODE_POLICY_MANAGER", dump_folder=log_dir)
+
+        self.allocate_trainers()
+
+    def allocate_trainers(self):
+        self._policy2trainer = defaultdict(list)
+        self._trainer2policies = defaultdict(list)
+
+        # initialize
+        if len(self._num_experiences_by_policy) == 0:
+            if len(self.policy_dict) >= self.num_trainers:
+                for i, name in enumerate(self.policy_dict):
+                    trainer_id = i % self.num_trainers
+                    self._policy2trainer[name].append(f"TRAINER.{trainer_id}")
+                    self._trainer2policies[f"TRAINER.{trainer_id}"].append(name)
+            else:
+                trainer_id_list = list(range(self.num_trainers))
+                for i, name in enumerate(self.policy_dict):
+                    for trainer_id in trainer_id_list[i::len(self.policy_dict)]:
+                        self._policy2trainer[name].append(f"TRAINER.{trainer_id}")
+                        self._trainer2policies[f"TRAINER.{trainer_id}"].append(name)
+
+        # allocate trainers according to historical experience numbers.
```

Comment:
Other allocation mechanisms are also possible. We may need to abstract the allocation logic.